### PR TITLE
PEP 8: Add note regarding exception message string literals

### PR DESCRIPTION
**Pull request by erlend-aasland**
Resolves #2604

### CODE REVIEW

File: pep-0008.txt
Comment (erlend-aasland) [inline pep-0008.txt:246]: Maybe there's a better place to put this into PEP 8, but this was my first thought after skimming through.

File: pep-0008.txt
Comment (AA-Turner) [inline pep-0008.txt:250]: This could be a link, but equally it follows the line length section directly

**Review by AA-Turner — APPROVED** [review]
Location in the text seems fine to me.

A

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt:250]: I'm not sure it's needed, but I'm also fine with a link. I'll leave it for you and Christopher to decide 🙂

(IIRC, both of you have the commit bit here, so feel free to apply the suggestion when you've decided.)

File: pep-0008.txt
Comment (CAM-Gerlach) [reply pep-0008.txt:246]: Seems like the best spot to me :+1:

File: pep-0008.txt
Comment (CAM-Gerlach) [reply pep-0008.txt:250]: > I'm not sure it's needed, but I'm also fine with a link. 

I'm not sure it is either, since it is a short section and immediately above. But I don't have a strong opinion.

> I'll leave it for you and Christopher to decide slightly_smiling_face

You mean @Rosuav ? Or are you referring to me (CAM)?

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @erlend-aasland

**Comment by gvanrossum** [comment]
@warsaw What do you think? It’s a bit of a u-turn but I think grepping for errors would be more effective.

**Review by gvanrossum — COMMENTED** [review]
Definitely wait for Barry to review.

I wonder if we should generalize to "error message"?

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt:250]: I was referring to you, CAM :)

**Comment by erlend-aasland** [comment]
> Definitely wait for Barry to review.

Sure. I should've done the same for PEP 7; sorry 'bout the rushed merge.

> I wonder if we should generalize to "error message"?

+1

File: pep-0008.txt
Comment (AA-Turner) [reply pep-0008.txt]: > I was unaware that there was more than two PEP editors

Guido, Barry, Bret, ChrisA, Jelle, Adam, Cam, and Hugo are the current complement -- a few more than two!

A

Comment (Rosuav) [reply pep-0008.txt]: It's okay, most of us are invisible :) For what it's worth, I have no opinion on the specific wording, but I fully approve of the sentiment here.

**Comment by warsaw** [comment]
> What do you think? It’s a bit of a u-turn but I think grepping for errors would be more effective.

Agreed

> I wonder if we should generalize to "error message"?

+1

I guess I'm +0 on this change for PEP 8.  We have to be careful not to overload PEP 8 for every possible corner case, but I agree with @gvanrossum on greppability.

Let me quibble just a bit with the wording.

File: pep-0008.txt
Comment (warsaw) [inline pep-0008.txt:245]: I think just saying "Error messages" would be sufficient.  I have to pause to think about what "Error message string literals" are.  Exception strings?  Something else?  Is there something special about string literals for errors?

File: pep-0008.txt
Comment (warsaw) [inline pep-0008.txt:248]: s/error message string literals/error messages/

File: pep-0008.txt
Comment (warsaw) [inline pep-0008.txt:251]: Maybe we don't need the "or just bend that rule." and the invocation of practicality over purity?  All of that should be well understood if you read PEP 8.

**Review by warsaw — CHANGES_REQUESTED** [review]
Wording suggestions.

File: pep-0008.txt
Comment (CAM-Gerlach) [reply pep-0008.txt:245]: My one concern with that is that readers might think its talking about the wording/formatting of error messages themself, rather than concerned with how string literals for error messages are formatted in the code. But I can see both sides of this.

File: pep-0008.txt
Comment (CAM-Gerlach) [reply pep-0008.txt:251]: I suppose the benefit of making it explicit is that it establishes that preserving greppability should generally take priority over following the line length restriction, which is otherwise left ambigious by the current wording and thus would seem to neuter much of the intent of the addition here, perhaps.

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt:245]: How about just dropping "Literals"?

"Strive to keep error message strings on one line"

vs.

"Strive to keep error messages on one line"

**Comment by erlend-aasland** [comment]
> I wonder if we should generalize to "error message"?

FWIW, I'm inclined to _not_ applying this generalisation to the PEP 7 change (#2605).

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt:251]: I prefer the explicity of "or just bend that rule", but I will remove it if there is a strong opposition to it.

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt]: PTAL; I've removed "literals".

File: pep-0008.txt
Comment (erlend-aasland) [reply pep-0008.txt]: Can I mark this as resolved? Did you decide about the link?

**Comment by warsaw** [comment]
I've thought about this over the weekend.  @gvanrossum I don't know, I kind of think maybe this and the PEP 7 changes *shouldn't* go in.  It's seems vague and less focused on making code readable than the effects of that code.  PEP 8 is already huge, so I think the bar has to be pretty high on the "this helps readability" scale.

**Comment by erlend-aasland** [comment]
> I've thought about this over the weekend. @gvanrossum I don't know, I kind of think maybe this and the PEP 7 changes _shouldn't_ go in. It's seems vague and less focused on making code readable than the effects of that code. PEP 8 is already huge, so I think the bar has to be pretty high on the "this helps readability" scale.

Ok. Again, sorry for not waiting on your review. I'll be more careful next time.

I'll prepare a PR for reverting this.

**Comment by gvanrossum** [comment]
In the face of Barry's strong opinion I back out on updating the PEPs. I still think we should generally not be breaking up error messages across lines.

**Comment by erlend-aasland** [comment]
> I still think we should generally not be breaking up error messages across lines.

Ditto

**Pull request closed without merging**
